--- OpenColorIO-1.1.0/src/core/OCIOYaml.cpp.orig	2018-01-05 02:38:27.000000000 +0100
+++ OpenColorIO-1.1.0/src/core/OCIOYaml.cpp	2018-06-25 21:49:48.951035161 +0200
@@ -30,7 +30,7 @@
 
 #include <OpenColorIO/OpenColorIO.h>
 
-#ifndef WIN32
+#if 0
 
 // fwd declare yaml-cpp visibility
 #pragma GCC visibility push(hidden)
--- OpenColorIO-1.1.0/CMakeLists.txt.orig	2018-06-26 05:31:36.084052127 +0200
+++ OpenColorIO-1.1.0/CMakeLists.txt	2018-06-26 19:16:47.136820152 +0200
@@ -271,7 +271,7 @@
     mark_as_advanced(YAML_CPP_INCLUDE_DIR YAML_CPP_LIBRARY YAML-CPP_FOUND)
 
     if(YAML_CPP_FOUND)
-        if(YAML_CPP_VERSION VERSION_GREATER "0.5.0")
+        if(YAML_CPP_VERSION VERSION_GREATER "0.5.0" AND YAML_CPP_VERSION VERSION_LESS "0.6.0")
             # Need to also get the boost headers here, as yaml-cpp 0.5.0+ requires them.
             # Don't bother doing this step if we are already including the boost headers for shared_ptr
             if(NOT OCIO_USE_BOOST_PTR)
