--- OpenColorIO-1.1.0/src/core/CMakeLists.txt.orig	2018-06-25 19:35:05.841127458 +0200
+++ OpenColorIO-1.1.0/src/core/CMakeLists.txt	2018-06-25 19:59:35.564444007 +0200
@@ -94,7 +94,7 @@
         VERSION ${OCIO_VERSION}
         SOVERSION ${SOVERSION})
     
-    install(TARGETS OpenColorIO_STATIC EXPORT OpenColorIO ARCHIVE DESTINATION ${CMAKE_INSTALL_EXEC_PREFIX}/lib/static)
+    install(TARGETS OpenColorIO_STATIC EXPORT OpenColorIO ARCHIVE DESTINATION ${CMAKE_INSTALL_EXEC_PREFIX}/lib${LIB_SUFFIX})
 endif()
 
 macro(target_link_OCIO target)
