--- OpenColorIO-1.1.0/CMakeLists.txt.orig	2018-06-26 19:16:47.136820152 +0200
+++ OpenColorIO-1.1.0/CMakeLists.txt	2018-06-26 19:59:46.266790685 +0200
@@ -460,7 +460,11 @@
 if(OCIO_BUILD_APPS AND (OCIO_BUILD_STATIC OR OCIO_BUILD_SHARED) )
 
     # Try to find OpenImageIO (OIIO) and OpenGL stuff
+    if(DISABLE_OIIO)
+        set(OIIO_FOUND FALSE)
+    else()
     OCIOFindOpenImageIO()
+    endif()
     
     if(OIIO_FOUND)
         add_subdirectory(src/apps/ocioconvert)
